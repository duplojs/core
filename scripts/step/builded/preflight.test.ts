import { Process } from "@scripts/duplose/process";
import { type ProcessStepParams } from "../process";
import { makeFloor } from "@scripts/floor";
import { BuildedPreflightStep } from "./preflight";
import { PreflightStep } from "../preflight";
import { duploTest } from "@test/utils/duploTest";

describe("BuildedPreflightStep", () => {
	const process = new Process("test");
	process.instance = duploTest;
	process.setOptions({
		toto: 1,
		test: "",
	});
	process.setInput(22);

	it("merge object options", () => {
		const params: ProcessStepParams = {
			options: {
				toto: 2,
				test1: "&",
			},
		};

		const step = new PreflightStep(process, params);

		const buildedProcessStep = new BuildedPreflightStep(duploTest, step);

		expect(buildedProcessStep.params.options).toStrictEqual({
			toto: 2,
			test: "",
			test1: "&",
		});

		expect(buildedProcessStep.params.input?.(makeFloor().pickup)).toBe(22);
	});

	it("merge function options", () => {
		const params: ProcessStepParams = {
			options: () => ({
				toto: 2,
				test1: "&",
			}),
		};

		const step = new PreflightStep(process, params);

		const buildedProcessStep = new BuildedPreflightStep(duploTest, step);

		expect(
			typeof buildedProcessStep.params.options === "function"
				? buildedProcessStep.params.options(makeFloor().pickup)
				: undefined,
		).toStrictEqual({
			toto: 2,
			test: "",
			test1: "&",
		});
	});

	it("toString options: function, pickup", async() => {
		const params: ProcessStepParams = {
			pickup: ["rr", "tt"],
			options: () => ({
				toto: 2,
				test1: "&",
			}),
		};

		const step = new PreflightStep(process, params);

		const buildedProcessStep = new BuildedPreflightStep(duploTest, step);

		await expect(buildedProcessStep.toString(1)).toMatchFileSnapshot("__data__/preflight1.txt");
	});

	it("toString options: object, input, skip", async() => {
		const params: ProcessStepParams = {
			input: () => undefined,
			options: {
				toto: 2,
				test1: "&",
			},
			skip: () => true,
		};

		const step = new PreflightStep(process, params);

		const buildedProcessStep = new BuildedPreflightStep(duploTest, step);

		await expect(buildedProcessStep.toString(1)).toMatchFileSnapshot("__data__/preflight2.txt");
	});
});
