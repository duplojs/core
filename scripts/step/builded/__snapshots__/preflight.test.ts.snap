// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildedPreflightStep > toString options: function, pickup 1`] = `
"
/* beforePreflightStep(index: 1) */
/* end_block */


				result = this.preflightSteps[1].processFunction(
					request,
					this.preflightSteps[1].params.options(floor.pickup),
					this.preflightSteps[1].params.input(floor.pickup)
				);

				
/* beforeTreatResultPreflightStep(index: 1) */
/* end_block */


				if(result instanceof this.Response){

break Execution;
}

				
/* beforeIndexingResultPreflightStep(index: 1) */
/* end_block */


				floor.drop("rr", result["rr"]);
floor.drop("tt", result["tt"]);
				

/* afterPreflightStep(index: 1) */
/* end_block */
"
`;

exports[`BuildedPreflightStep > toString options: object, input, skip 1`] = `
"
/* beforePreflightStep(index: 1) */
/* end_block */


		if(!this.preflightSteps[1].params.skip(floor.pickup)){
			
				result = this.preflightSteps[1].processFunction(
					request,
					this.preflightSteps[1].params.options,
					this.preflightSteps[1].params.input(floor.pickup)
				);

				
/* beforeTreatResultPreflightStep(index: 1) */
/* end_block */


				if(result instanceof this.Response){

break Execution;
}

				
/* beforeIndexingResultPreflightStep(index: 1) */
/* end_block */


				
				
		}
		

/* afterPreflightStep(index: 1) */
/* end_block */
"
`;
